src/ucioption.h

```diff
@@ -72,10 +55,43 @@ class Option {
     friend std::ostream& operator<<(std::ostream&, const OptionsMap&);
 
    private:
-    std::string defaultValue, currentValue, type;
-    int         min, max;
-    size_t      idx;
-    OnChange    on_change;
+    friend class OptionsMap;
+    friend class Engine;
+    friend class Tune;
+
+    void operator<<(const Option&);
+
+    std::string       defaultValue, currentValue, type;
+    int               min, max;
+    size_t            idx;
+    OnChange          on_change;
+    const OptionsMap* parent = nullptr;
+};
+
+class OptionsMap {
```

Comment:
Since Options hold a backpointer to OptionsMap it would be wise to make OptionsMap immovable. So just like the Engine class. I don't think it's important for it to be movable so we can spare pointer rewiring complexity.